### DISCUSSION TITLE
Fabric <- (Mill) -> Obsidian

### DISCUSSION THREAD

**Discussion by nic0711** [danielmiessler/Fabric#131] [Ideas]
Hi!
It would be great if it were possible to combine Fabric with Obsidian.
For example to:
a) edit / categorize / ... the patterns in Obsidian.
b) store the output of Fabric in the Obsidian "brain" and make it usable in context

I am sure that there are many more possible applications...

E.g. a "Mill pipeline" to the Obsidian vault - wouldn't that make Fabric "instantly" available on mobile? ;)

What do you think?

**Comment by smrki1** [comment]
That is already possible is it not? You can add your patterns into a obsidian vault and edit them there with obsidian if you want, as they are just a text file. You can also send your output wherever you want with your command line interface.

**Reply by nic0711** [reply]
I'm just thinking: how is that on mobile... hence my thought about the "Mill". 
Let's say I can edit the patterns on my mobile - then I'd like to be able to also run them on mobile. ;)

For that, it would be nice to have Fabric on a private server / RasPy / Synology / ... and then return the result to me in Obsidian, for example. 
... that's how I understood the "Mill". 
Maybe I am wrong.

**Comment by danielmiessler** [comment]
I think the issue is running them on mobile. It sounds like there you might need a publicly available server to hit. I'd have to think about it more.

**Reply by nic0711** [reply]
Hi Daniel!
Yes, a server is needed. Therefore i thought about the rasPy / Synology (or vserver) approach. 

It might also work dockerized on a vserver - this might also help with the Obsidian vault... may be. ;)

**Reply by nic0711** [reply]
... or a web frontend

**Comment by TomLucidor** [comment]
Why just Obsidian and not Notion... or LogSeq for FOSS?

**Comment by evancris** [comment]
There is a Fabric Plugin for Obsidian that works nicely for a local installation. 

unofficial-fabric-plugin 
https://github.com/chasebank87/unofficial-fabric-plugin

**Reply by usama-siala** [reply]
i also checked mesh-ai plugin , it is still beta, but it works great.

**Comment by davejpeters** [comment]
If you're comfortable with it you can use the "Termius" app to SSH to your machine from your phone. You can then use the command line as if you were at your desktop. I use it out and about when I have a query to send through fabric. When I get home, if I've saved the outputs to my Obsidian Vault, all the notes will be there in my obsidian vault. If I leave the Vault in sync mode while I'm out, the results from fabric will be synced to my mobile Obsidian.

**Reply by JuracyAmerico** [reply]
Hi @johnconnor-sec , wow, this is very interesting, could you please give more details how you archived that?

**Reply by davejpeters** [reply]
- [ ] Download the [Tailscale](https://github.com/tailscale/tailscale) app on your 'fabric' machine and iOS. Sign up at [Tailscale](https://tailscale.com).
- [ ] Add your 'fabric' machine and your mobile device to the Tailscale overlay network. 
- [ ] Download the [Termius](https://termius.com/free-ssh-client-for-iphone) app on your device. 
- [ ] Connect the Termius app to your 'fabric machine' via SSH on your local network. 
- [ ] Ensure Tailscale is up on both the 'fabric machine' and your device. 
- [ ] Test outside your local network and on the road.

It has been a while since I set this up and I'm just going off memory, but if I remember correctly then this should work with a bit of tweaking. Tailscale + Termius will allow you to SSH into your machine from outside your home network and use the command line, essentially allowing access to fabric along with any other CLI apps that you have. Pipe the output of fabric into your Obsidian notebook using the `>` operator or setup fabric to automatically add outputs to your notebook.
